rdrf/rdrf/templates/rdrf_cdes/form.html

```diff
@@ -540,24 +539,188 @@
 
  </div>
 
-
+    <div id="sessionTimeoutModal" class="modal" tabindex="-1">
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-body">
+                    <p data-session-state="expiring">{% trans "Your session will expire in " %}<strong id="sessionExpiryTimeout">X seconds</strong>.</p>
+                    <p data-session-state="expired" class="d-none alert alert-danger">{% trans "Your session has expired." %}</p>
+                </div>
+                <div class="modal-footer">
+                    <div data-session-state="expiring">
+                        <button id="continueWorkingBtn" type="button" class="btn btn-primary" data-bs-dismiss="modal">{% trans "Continue working" %}</button>
+                        <button id="logoutBtn" type="button" class="btn btn-secondary" data-bs-dismiss="modal">{% trans "Logout" %}</button>
+                    </div>
+                    <div data-session-state="expired" class="d-none">
+                        <button id="goToLoginBtn" type="button" class="btn btn-primary" data-bs-dismiss="modal">{% trans "Go to Login Page" %}</button>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
 <script>
+
+    function ModalSessionNotifier() {
+
+        const SessionState = {
+            Expiring: "expiring",
+            Expired: "expired"
+        }
+
+        const $sessionTimeoutModal = $('#sessionTimeoutModal');
+        const $sessionExpiredComponents = $(`[data-session-state="${SessionState.Expired}"]`);
+        const $sessionExpiringComponents = $(`[data-session-state="${SessionState.Expiring}"]`);
+        const sessionTimeoutModal = new bootstrap.Modal($sessionTimeoutModal, {backdrop: "static"});
+        let countdownInterval;
+
+        function updateSessionTimeoutMessage(secondsLeft) {
+            const ONE_MINUTE_IN_SECONDS = 60;
+            let timeoutText = "";
+
+            if (secondsLeft >= ONE_MINUTE_IN_SECONDS) {
+                const minutesLeft = Math.round(secondsLeft / ONE_MINUTE_IN_SECONDS);
+                timeoutText = interpolate(ngettext("%s minute", "%s minutes", minutesLeft), [minutesLeft]);
```

Comment:
It appears the string is not picked up in the`makemessages` part of the  `build_translation_sources`  script. The JS might need to be partially moved to a static file.

Reply:
Done.
```
#: rdrf/rdrf/static/js/session_timeout_manager.js:25
#, javascript-format
msgid "%s minute"
msgid_plural "%s minutes"
msgstr[0] ""
msgstr[1] ""

#: rdrf/rdrf/static/js/session_timeout_manager.js:27
#, javascript-format
msgid "%s second"
msgid_plural "%s seconds"
msgstr[0] ""
msgstr[1] ""
```